packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -816,6 +816,81 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
         }
     }
 
+    /**
+     * @inheritdoc IAssetManager
+     */
+    function claimKghReward(address validator, uint256[] calldata tokenIds) external {
+        if (validator == address(0)) revert ZeroAddress();
+        if (tokenIds.length == 0) revert NotAllowedZeroInput();
+
+        Vault storage vault = _vaults[validator];
+        KghDelegator storage kghDelegator = vault.kghDelegators[msg.sender];
+        if (kghDelegator.kghNum < tokenIds.length) revert InvalidTokenIdsInput();
+
+        // Auto compounding is not applied for claim function.
+        uint128 claimedRewards = _claimBoostedReward(validator, msg.sender, tokenIds.length, false);
```

Comment:
Please add revert if claimedRewards is 0. Base reward cannot be more than 0 if boosted reward is 0